lib/wiki_pageviews.rb

```diff
@@ -101,13 +101,23 @@ def parse_results(response)
     return data['items'] if data['items']
     # As of October 2017, the data type is https://www.mediawiki.org/wiki/HyperSwitch/errors/not_found
     return no_results if %r{errors/not_found}.match?(data['type'])
+    return no_results if no_data_available_response?(data)
     raise PageviewApiError, response
   end
 
   def no_results
     {}
   end
 
+  # As of October 2023, we started to see 404 not found responses with about:blank type
+  # and a specific detail when handling requests for which no data is available
+  def no_data_available_response?(response)
+    no_data_avialable_detail = 'The date(s) you used are valid, but we either do not have data '\
```

Comment:
I think relying on the exact 'detail' message here is going to be too fragile. Is this always the message when type is `about:blank`? I think the `type` is less likely to change than the text of the 'detail' message, which could get updated wording at any point.

Reply:
At first glance, I thought that `about:blank`as `type` was a kind of bug on the API side, as I didn't find any specific documentation for that and it wasn't pretty meaningful for me. That's why I decided to rely on the `detail` instead, I thought that maybe the `type`  would change in the near future. However, I think we can rely on the `status` as you mentioned in your other comment.